src/Tools/SeleniumGenerator/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DotVVM.Framework.Tools.SeleniumGenerator.Extensions
+{
+    public static class DictionaryExtensions
+    {
+        public static Dictionary<TKey, TValue> AddRange<TKey, TValue>(this Dictionary<TKey, TValue> first,
+            Dictionary<TKey, TValue> second)
+        {
+            return first.Union(second).ToDictionary(t => t.Key, t => t.Value);
+        }
+    }
+    public static class ServiceProviderExtensions
+    {
+        public static T TryGetService<T>(this IServiceProvider serviceProvider)
```

Comment:
Is this needed? GetService will return null by default when the service is not registred.